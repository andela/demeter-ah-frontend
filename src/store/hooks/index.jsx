import { useEffect } from 'react';
import { getUser } from '../actions/signup';

export const useSetUser = ({ dispatch }) => {
  useEffect(() => {
    dispatch(getUser());
  }, []);
  return [];
};
