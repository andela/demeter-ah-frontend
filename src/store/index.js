import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import combineReducers from './reducers';
import dummyAction from './actions/Signup';

const middlewares = applyMiddleware(thunk);

const store = createStore(
  combineReducers,
  composeWithDevTools(middlewares),
);

store.dispatch(dummyAction);

export default store;
