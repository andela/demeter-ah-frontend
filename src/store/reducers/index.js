import { combineReducers } from 'redux';
import auth from './Auth';
import articleListing from './articleListing';
import resetPassword from './resetPassword';
import changePassword from './changePassword';
import articles from './articles';
import editProfile from './editProfile';
import viewProfile from './viewProfile';
import membership from './Membership';
import viewArticle from './viewArticle';
import relatedArticles from './relatedArticles';
import bookmarks from './bookmarks';
import bookmarkArticle from './bookmarkArticle';
import comments from './Comments';
import search from './search';
import notifications from './notifications';
import rate from './rateArticle';

const reducers = combineReducers({
  auth,
  resetPassword,
  changePassword,
  articles,
  editProfile,
  viewProfile,
  membership,
  articleListing,
  viewArticle,
  relatedArticles,
  bookmarks,
  bookmarkArticle,
  comments,
  search,
  notifications,
  rate
});

export default reducers;
