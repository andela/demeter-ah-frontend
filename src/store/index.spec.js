import store from './index';

describe('STORE', () => {
  test('Test store', () => {
    expect(store).toMatchSnapshot();
  });
});
