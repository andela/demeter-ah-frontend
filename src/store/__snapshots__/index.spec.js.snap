// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`STORE Test store 1`] = `
Object {
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
  Symbol(observable): [Function],
}
`;
